File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/entity/AssetType.java
package com.NBE3_4_2_Team4.domain.asset.main.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.Getter;

@Getter
public enum AssetType {
    CASH("캐시"),
    POINT("포인트"),
    ALL("전체");

    private final String displayName;

    AssetType(String name) {
        this.displayName = name;
    }

    @JsonValue
    @Override
    public String toString() {
        return this.displayName;
    }

    @JsonCreator
    public static AssetType fromValue(String value) {
        for (AssetType assetType : values()) {
            if (assetType.name().equalsIgnoreCase(value) || assetType.getDisplayName().equalsIgnoreCase(value)) {
                return assetType;
            }
        }
        return AssetType.ALL;
    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/entity/AssetType.kt
package com.NBE3_4_2_Team4.domain.asset.main.entity

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import lombok.Getter


enum class AssetType(val displayName: String) {
    CASH("캐시"),
    POINT("포인트"),
    ALL("전체");

    @JsonValue
    override fun toString(): String {
        return this.displayName
    }

    companion object {
        @JvmStatic
        @JsonCreator
        fun fromValue(value: String?): AssetType {
            for (assetType in entries) {
                if (assetType.displayName.equals(value, ignoreCase = true) || assetType.name
                        .equals(value, ignoreCase = true)
                ) {
                    return assetType
                }
            }
            return ALL
        }
    }
}